### PR TITLE
Declarative pipeline for the monorepo elastic/beats.git

### PR DESCRIPTION
**Pull request by v1v**
### What

Even though https://github.com/elastic/beats/blob/7.8/Jenkinsfile is not anymore something we use, it's a valid complex declarative pipeline with several stages that we were using in our production environment [here](https://beats-ci.elastic.co/job/Beats/job/beats/).

cc @kuisathaverat

### CODE REVIEW

File: data/valid/monorepo-with-shared-library-steps/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,1391 @@
+#!/usr/bin/env groovy
```
Comment (v1v) [inline data/valid/monorepo-with-shared-library-steps/Jenkinsfile]: This is not what we want to evaluate, so I'll remove this shebang

Comment (v1v) [reply data/valid/monorepo-with-shared-library-steps/Jenkinsfile]:
```suggestion
```

**Comment by rtyler** [comment]
Oof! This is definitely a challenging one, thanks for the contribution!

I was wondering how soon somebody would come along with a Declarative Pipeline that had Groovy Script hiding in the eaves, turns out, immediately :smiling_imp:

**Comment by v1v** [comment]
:)

**Comment by kuisathaverat** [comment]
@v1v out of my head!!!! 😄

**Comment by rtyler** [comment]
So this `Jenkinsfiles` can be parsed, but I still have work to do in order to have the `jdp` CLI point out when there's scary potentially invalid Groovy hiding outside the `pipeline {}`

**Pull request merged**